### PR TITLE
Backport 2.28: Make check_test_cases.py recognize test case name templates in ssl-opt.sh

### PR DESCRIPTION
**Pull request by tgonzalezorlandoarm**
## Description

Backport of #8088 

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required
- [x] **backport** This is the backport
- [x] **tests** provided

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

File: tests/compat.sh [hunk 1]
```diff
@@ -142,7 +142,7 @@ print_test_case() {
 }
 
 # list_test_case lists all potential test cases in compat.sh without execution
-list_test_case() {
+list_test_cases() {
```
Comment (gilles-peskine-arm) [inline tests/compat.sh:145]: Please update this in https://github.com/Mbed-TLS/mbedtls/pull/8088 as well. The spelling of the function name doesn't matter much, but if the branches diverge, it makes future backports harder.

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Comment by gilles-peskine-arm** [comment]
Unfortunately there's a merge conflict in 2.28 as well.

**Comment by tgonzalezorlandoarm** [comment]
> Unfortunately there's a merge conflict in 2.28 as well.

Done

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Review by daverodgman — APPROVED** [review]

**Comment by daverodgman** [comment]
Manually merged since the DCO check was stuck. CI was passing & merge queue was empty so no risk of collisions.

**Pull request merged**